providers/universal-provider/src/providers/eip155.ts

```diff
@@ -198,6 +201,41 @@ class Eip155Provider implements IProvider {
     }
     return capabilities;
   }
+
+  private async getCallStatus(args: RequestParams) {
+    const session = this.client.session.get(args.topic);
+    if (
+      session?.sessionProperties &&
+      Object.keys(session.sessionProperties).includes("bundler_url")
+    ) {
+      const bundlerUrl = session.sessionProperties.bundler_url;
+      try {
+        return await this.getUserOperationReceipt(bundlerUrl, args);
+      } catch (error) {
+        console.warn("Failed to fetch call status from bundler", error, bundlerUrl);
+      }
+    }
+    if (this.namespace.methods.includes(args.request.method)) {
+      return await this.client.request(args as EngineTypes.RequestParams);
+    }
+
+    throw new Error("Fetching call status not approved by the wallet.");
+  }
+
+  private async getUserOperationReceipt(bundlerUrl: string, args: RequestParams) {
+    const url = new URL(bundlerUrl);
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(
+        formatJsonRpcRequest("eth_getUserOperationReceipt", [args.request.params?.[0]]),
+      ),
+    });
+
+    return await response.json();
```

Comment:
Consider checking if the response is successful before parsing it as a json.

```suggestion
    if(!response.ok){
      throw new Error(`Failed to fetch user operation receipt - ${response.status}`)
    }
    return await response.json();
```